ddtrace/mocktracer/mocktracer_test.go

```diff
@@ -74,9 +74,14 @@ func TestTracerStartSpan(t *testing.T) {
 
 func TestTracerFinishedSpans(t *testing.T) {
 	var mt mocktracer
+	assert.Empty(t, mt.FinishedSpans())
+	assert.Empty(t, mt.OpenSpans())
 	parent := newSpan(&mt, "http.request", &ddtrace.StartSpanConfig{})
 	child := mt.StartSpan("db.query", tracer.ChildOf(parent.Context()))
+	assert.Empty(t, mt.FinishedSpans())
+	assert.Len(t, mt.OpenSpans(), 1)
 	child.Finish()
+	assert.Empty(t, mt.OpenSpans())
```

Comment:
Question for you @gbbr , should the parent span get included in `openSpans`? Currently it will not, which causes an asymmetry with `finishedSpans`, since it does get added to that collection once it is closed.

Reply:
I don't understand the question. Any started span that is unfinished should be considered open. Perhaps you can explain in more detail?

Reply:
This test illustrates the problem. `parent` and `child` get created, but only `child` goes through `mocktracer`'s `StartSpan()` method. At that point, `FinishedSpans()` is empty, and `OpenSpans()` only contains `child`. Once both spans get finished, `FinishedSpans()` contains both spans. At the beginning, `len(OpenSpans()) == 1`. At the end, `len(FinishedSpans()) == 2`.

Reply:
Perhaps the answer here is that `parent` should also be initialized via `mt.StartSpan()`, instead of calling `newSpan` directly. Thoughts?

Reply:
> Perhaps the answer here is that parent should also be initialized via mt.StartSpan(), instead of calling newSpan directly.

Exactly!